cmd/main.go

```diff
@@ -1,34 +1,41 @@
 package main
 
 import (
-	"flag"
 	"fmt"
 	"os"
+	"strings"
 	"weather"
 )
 
-type Response struct {
-	OneWord string  `json:"oneword"`
-	Celcius float64 `json:"celcius"`
+func LocationFromArgs(args []string) string {
+
+	var command strings.Builder
+
+	for i, v := range os.Args {
+		if i > 0 {
+			command.WriteString(v)
+			command.WriteString(" ")
+		}
+	}
+
+	return command.String()
 }
 
 func main() {
 
-	location := flag.String("location", "", "a city")
-	flag.Parse()
-
-	if len(*location) == 0 {
-		fmt.Printf("please enter a location\n")
+	if len(os.Args) == 1 {
```

Comment:
Since we now have a function that turns arguments into a location, it seems like this error-checking would be better done there. You could have `LocationFromArgs` return this error message if there isn't a location specified, and then all `main` needs to do is print it out.